core/iwasm/libraries/debug-engine/debug_engine.c

```diff
@@ -987,6 +1016,38 @@ wasm_debug_instance_interrupt_all_threads(WASMDebugInstance *instance)
     return true;
 }
 
+bool
+wasm_debug_instance_detach(WASMDebugInstance *instance)
+{
+    WASMExecEnv *exec_env;
+
+    if (!instance)
+        return false;
+
+    exec_env = bh_list_first_elem(&instance->cluster->exec_env_list);
+    if (!exec_env)
+        return false;
+
+    wasm_gdbserver_detach(instance->control_thread->server);
+
+    while (exec_env) {
+        if (instance->current_state == APP_STOPPED) {
+            /* Resume all threads since remote debugger detached*/
+            os_mutex_lock(&exec_env->wait_lock);
+            exec_env->current_status->running_status = STATUS_RUNNING;
+            os_cond_signal(&exec_env->wait_cond);
+            os_mutex_unlock(&exec_env->wait_lock);
+        }
+        exec_env = bh_list_elem_next(exec_env);
+    }
+
+    /* relaunch, and stop the thread*/
```

Comment:
```suggestion
    /* relaunch, and stop the thread */
```

Reply:
does it really stop the thread? Seems the thread still exists, just block on accept again?
Had better change to "relaunch, accept new debug connection"